src/cdk/v2/destinations/bluecore/utils.js

```diff
@@ -0,0 +1,248 @@
+        );
+      }
+      if (
+        !isDefinedAndNotNull(payload?.properties?.customer) ||
+        Object.keys(payload.properties.customer).length === 0
+      ) {
+        throw new InstrumentationError(
+          `[Bluecore] property:: No relevant trait to populate customer information, which is required for ${payload.event} action`,
+        );
+      }
+      break;
+    default:
+      break;
+  }
+};
+
+/**
+ * Deduces the track event name based on the provided track event name and configuration.
+ *
+ * @param {string} trackEventName - The track event name to deduce.
+ * @param {object} Config - The configuration object.
+ * @returns {string|array} - The deduced track event name.
+ */
+const deduceTrackEventName = (trackEventName, Config) => {
+  let eventName;
+  const { eventsMapping } = Config;
+  validateEventName(trackEventName);
+  /*
+    Step 1: Will look for the event name in the eventsMapping array if mapped to a standard bluecore event.
+            and return the corresponding event name if found.
+     */
+  if (eventsMapping.length > 0) {
+    const keyMap = getHashFromArrayWithDuplicate(eventsMapping, 'from', 'to', false);
+    eventName = keyMap[trackEventName];
+  }
+  if (isDefinedAndNotNullAndNotEmpty(eventName)) {
+    const finalEvent = typeof eventName === 'string' ? [eventName] : [...eventName];
+    return finalEvent;
+  }
+
+  /*
+    Step 2: To find if the particular event is amongst the list of standard
+            Rudderstack ecommerce events, used specifically for Bluecore API
+            mappings.
+    */
+
+  const eventMapInfo = EVENT_NAME_MAPPING.find((eventMap) => {
+    if (eventMap.src.includes(trackEventName.toLowerCase())) {
+      return eventMap;
+    }
+    return false;
+  });
+  if (isDefinedAndNotNull(eventMapInfo)) {
+    return [eventMapInfo.dest];
+  }
+
+  // Step 3: if nothing matches this is to be considered as a custom event
+  return [trackEventName];
+};
+
+/**
+ * Determines if the given event name is a standard Bluecore event.
+ *
+ * @param {string} eventName - The name of the event to check.
+ * @returns {boolean} - True if the event is a standard Bluecore event, false otherwise.
+ */
+const isStandardBluecoreEvent = (eventName) => {
+  const standardEventList = EVENT_NAME_MAPPING.map((item) => item.dest);
+  return !!standardEventList.includes(eventName);
+};
+
+/**
+ * Adds an array of products to a message.
+ *
+ * @param {object} message - The message object to add the products to.
+ * @param {array|object} products - The array or object of products to add.
+ * @param {string} eventName - The name of the event.
+ * @throws {InstrumentationError} - If the products array is not defined or null.
+ * @returns {array} - The updated product array.
+ */
+const normalizeProductArray = (products) => {
+  let finalProductArray = null;
+  if (isDefinedAndNotNull(products)) {
+    const productArray = Array.isArray(products) ? products : [products];
```

Comment:
we have toArray function in https://github.com/rudderlabs/rudder-transformer/blob/b5d28a315e859f159dc4b58e547393e19da592db/src/util/common.js